packages/nextjs/src/server/clerkClient.ts

```diff
@@ -27,6 +34,63 @@ const clerkClient = createClerkClient({
     disabled: TELEMETRY_DISABLED,
     debug: TELEMETRY_DEBUG,
   },
+};
+
+export const clerkClientStorage = new AsyncLocalStorage<Partial<AuthenticateRequestOptions>>();
+
+const createClerkClientWithOptions: typeof createClerkClient = options =>
+  createClerkClient({ ...clerkClientDefaultOptions, ...options });
+
+const clerkClientSingleton = createClerkClient(clerkClientDefaultOptions);
+
+/**
+ * @deprecated
+ * This singleton is deprecated and will be removed in a future release. Please use `clerkClient()` as a function instead.
+ */
+const clerkClientSingletonProxy = new Proxy(clerkClientSingleton, {
```

Comment:
Calling out for JS help 🙇🏻 We want to log a deprecation warning only if `clerkClient` gets called as a singleton. 

The `Proxy` is being placed around the singleton reference here, but in line 90, it's properties are being merged with `clerkClient()` on `Object.assign`, therefore the deprecation warnings is being logged for both cases. 

It's important to keep exporting `clerkClient` and allow it to be either a singleton or a function but to do so, it requires that `Object.assign` bit - is there another way to do so while performing a proxy for the deprecation warning?

Reply:
Do you mean that the usage of `Object.assign()` is causing the deprecation warning to be logged? Or because we're using the proxied client object under the hood for the non-dynamic keys case, it logs for both:
```
clerkClient().users.getUser()

clerkClient.users.getUser()
```

Reply:
It's logging for both cases, and I think this is due to `Object.assign` since it's executing `Proxy.get` for every single property.